app/components/data-download.tsx

```diff
@@ -246,67 +237,91 @@ const DataDownloadInnerMenu = ({
   );
 };
 
+const DataDownloadMenuSection = ({
+  subheader,
+  dataToRender,
+  fileName,
+  onDownloaded,
+}: {
+  subheader: ReactNode;
+  dataToRender: PreparedData;
+  fileName: string;
+  onDownloaded: () => void;
+}) => {
+  return (
+    <>
+      <ListSubheader sx={{ mt: 3, lineHeight: 1.2, borderBottom: "none" }}>
+        {subheader}
+      </ListSubheader>
+      <MenuItem
+        sx={{
+          "&:hover": { backgroundColor: "transparent", cursor: "default" },
+        }}
+      >
+        <Flex sx={{ gap: 3 }}>
+          {FILE_FORMATS.map((fileFormat) => (
+            <DownloadMenuItem
+              key={fileFormat}
+              fileName={fileName}
+              fileFormat={fileFormat}
+              columnKeys={dataToRender.columnKeys}
+              data={dataToRender.data}
+              onDownloaded={onDownloaded}
+            />
+          ))}
+        </Flex>
+      </MenuItem>
+    </>
+  );
+};
+
 const DownloadMenuItem = ({
-  title,
+  fileName,
   data,
   columnKeys,
-  extent,
   fileFormat,
   onDownloaded,
 }: {
-  title: string;
+  fileName: string;
   data: Observation[];
   columnKeys: string[];
-  extent: Extent;
   fileFormat: FileFormat;
   onDownloaded: () => void;
 }) => {
   const [isDownloading, setIsDownloading] = useState(false);
   const download = useCallback(() => {
     return fetch("/api/download", {
       method: "POST",
-      headers: {
-        "Content-Type": "application/json",
-      },
-      body: JSON.stringify({
-        columnKeys,
-        data,
-        fileFormat,
-      }),
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify({ columnKeys, data, fileFormat }),
     }).then((res) =>
-      res.blob().then((blob) => saveAs(blob, `${title}.${fileFormat}`))
+      res.blob().then((blob) => saveAs(blob, `${fileName}.${fileFormat}`))
     );
-  }, [columnKeys, data, fileFormat, title]);
+  }, [columnKeys, data, fileFormat, fileName]);
 
   return (
-    <MenuItem
+    <Button
+      variant="text"
+      size="small"
+      endIcon={isDownloading ? <CircularProgress /> : null}
       onClick={async () => {
         setIsDownloading(true);
         await download();
         onDownloaded();
         setIsDownloading(false);
```

Comment:
Can you wrap download() and onDownloaded() inside a try, with the setIsDwonloading(false) inside a finally ? This way if there is an error, the spinner will be removed from the button.